sample-code/spring-app/src/test/java/com/sap/ai/sdk/app/controllers/GroundingTest.java

```diff
@@ -0,0 +1,146 @@
+package com.sap.ai.sdk.app.controllers;
+
+import static java.time.LocalDate.now;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import com.sap.ai.sdk.grounding.model.CollectionsListResponse;
+import com.sap.ai.sdk.grounding.model.DataRepositories;
+import com.sap.ai.sdk.grounding.model.DocumentResponse;
+import com.sap.ai.sdk.grounding.model.Documents;
+import com.sap.ai.sdk.grounding.model.DocumentsListResponse;
+import com.sap.ai.sdk.grounding.model.Pipelines;
+import com.sap.ai.sdk.grounding.model.RetievalSearchResults;
+import com.sap.cloud.sdk.services.openapi.core.OpenApiResponse;
+import java.time.format.TextStyle;
+import java.util.Locale;
+import java.util.UUID;
+import org.junit.jupiter.api.Test;
+
+class GroundingTest {
+  /** Java end-to-end test specific configuration ID. "name":"config-java-e2e-test" */
+  public static final String CONFIG_ID = "67e8d039-c7f1-4179-9f8f-60d158a36b0e";
+
+  private static final String JSON_FORMAT = "json";
+
+  @Test
+  void getPipelinesGetAll() {
+    final var controller = new GroundingController();
+
+    final var result = controller.getAllPipelines(JSON_FORMAT);
+    assertThat(result).isInstanceOf(Pipelines.class);
+    final var pipelinesList = ((Pipelines) result).getResources();
+    final var pipelinesCount = ((Pipelines) result).getCount();
+
+    // we don't have testable data yet, but the endpoint works without errors
```

Comment:
When will we get data?

Reply:
I created a backlog item for that.
JS colleagues also have no test yet.

https://github.com/SAP/ai-sdk-java-backlog/issues/202